presto-spark-base/src/test/java/com/facebook/presto/spark/execution/TestLocalTask.java

```diff
@@ -0,0 +1,466 @@
+            }
+            if (millisToIdleTarget < 0) {
+                return;
+            }
+            Thread.sleep(millisToIdleTarget);
+        }
+    }
+
+    private enum FailureScenario
+    {
+        NO_FAILURE,
+        TASK_MISMATCH,
+        TASK_MISMATCH_WHEN_VERSION_IS_HIGH,
+        REJECTED_EXECUTION,
+    }
+
+    @Path("/task/{nodeId}")
+    public static class TestingTaskResource
+    {
+        private static final UUID INITIAL_TASK_INSTANCE_ID = UUID.randomUUID();
+        private static final UUID NEW_TASK_INSTANCE_ID = UUID.randomUUID();
+
+        private final AtomicLong lastActivityNanos;
+        private final FailureScenario failureScenario;
+
+        private AtomicReference<TestingHttpClient> httpClient = new AtomicReference<>();
+
+        private TaskInfo initialTaskInfo;
+        private TaskStatus initialTaskStatus;
+        private long version;
+        private TaskState taskState;
+        private long taskInstanceIdLeastSignificantBits = INITIAL_TASK_INSTANCE_ID.getLeastSignificantBits();
+        private long taskInstanceIdMostSignificantBits = INITIAL_TASK_INSTANCE_ID.getMostSignificantBits();
+
+        private long statusFetchCounter;
+
+        public TestingTaskResource(AtomicLong lastActivityNanos, FailureScenario failureScenario)
+        {
+            this.lastActivityNanos = requireNonNull(lastActivityNanos, "lastActivityNanos is null");
+            this.failureScenario = requireNonNull(failureScenario, "failureScenario is null");
+        }
+
+        public void setHttpClient(TestingHttpClient newValue)
+        {
+            httpClient.set(newValue);
+        }
+
+        @GET
+        @Path("{taskId}")
+        @Produces(MediaType.APPLICATION_JSON)
+        public synchronized TaskInfo getTaskInfo(
+                @PathParam("taskId") final TaskId taskId,
+                @HeaderParam(PRESTO_CURRENT_STATE) TaskState currentState,
+                @HeaderParam(PRESTO_MAX_WAIT) Duration maxWait,
+                @Context UriInfo uriInfo)
+        {
+            lastActivityNanos.set(System.nanoTime());
+            return buildTaskInfo();
+        }
+
+        Map<PlanNodeId, TaskSource> taskSourceMap = new HashMap<>();
+
+        @POST
+        @Path("{taskId}")
+        @Consumes(MediaType.APPLICATION_JSON)
+        @Produces(MediaType.APPLICATION_JSON)
+        public synchronized TaskInfo createOrUpdateTask(
+                @PathParam("taskId") TaskId taskId,
+                TaskUpdateRequest taskUpdateRequest,
+                @Context UriInfo uriInfo)
+        {
+            for (TaskSource source : taskUpdateRequest.getSources()) {
+                taskSourceMap.compute(source.getPlanNodeId(), (planNodeId, taskSource) -> taskSource == null ? source : taskSource.update(source));
+            }
+            lastActivityNanos.set(System.nanoTime());
+            return buildTaskInfo();
+        }
+
+        public synchronized TaskSource getTaskSource(PlanNodeId planNodeId)
```

Comment:
not used; please check all unused methods in this class. Let's make sure the unit tests are purely for LocalTask. Otherwise, it just looks like a copy from another file.